### PR TITLE
support carma_wm_ctrl map change broadcast

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description

This PR gives laneletMap ability to remove individual regulatory element from a specific lanelet. Without it, regulatory element will be removed from the entire map, but we may want to keep its relation to some of the lanelets. For example, if regem (100) applied to both lanelet (101 and 102), now we can change regem of individual 101 or 102 lanelets.

## Related Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/726

## Motivation and Context
This PR supports changes in carma_wm_ctrl and carma_wm, of which PR addresses full flow of geofence request. This individual removal functionality was essential to avoid an edge case described above.

## How Has This Been Tested?

Existing and new unit tests pass
## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: lanelet2/lanelet2_core/src/LaneletMap.cpp [hunk 1]
```diff
@@ -774,6 +774,44 @@ void LaneletMap::add(Area area) {
...
+  if (regElem->id() == InvalId) {
+    throw InvalidInputError("Regulatory element with InvalId is passed to update()!");
+  }
+  else if (!regulatoryElementLayer.exists(regElem->id())) {
+    throw InvalidInputError("Id of the regulatory element is not registered in the map");
+  }
+ 
+  lanelet::Lanelets parent_llts = laneletLayer.findUsages(regElem);
+
+  if (regulatoryElementLayer.exists(regElem->id()) && std::find(parent_llts.begin(), parent_llts.end(), ll) == parent_llts.end()) {
```
Comment (msmcconnell) [inline lanelet2/lanelet2_core/src/LaneletMap.cpp]: The regulatoryElementLayer.exists(regElem->id()) check here is redundant given if statement above.

File: lanelet2/lanelet2_core/src/LaneletMap.cpp [hunk 2]
```diff
@@ -774,6 +774,44 @@ void LaneletMap::add(Area area) {
...
+    throw InvalidInputError("In remove function: The specified lanelet does not hold a regulatory element with that Id!");
+  }
+
+  // remove local copies of regem inside lanelets and their relationship in laneletlayer
+  for (Lanelet llt : laneletLayer.findUsages(regElem))
+  {
+    if (llt.id() == ll.id())
+    {
+      laneletLayer.remove(llt.id(), regElem);
+      llt.removeRegulatoryElement(regElem);
```
Comment (msmcconnell) [inline lanelet2/lanelet2_core/src/LaneletMap.cpp:804]: Do you think we should check if this lanelet is a parameter of the regulatory element as well and remove it if it is?

**Review by msmcconnell — CHANGES_REQUESTED** [review]

File: lanelet2/lanelet2_core/src/LaneletMap.cpp (hunk 2 above)
Comment (MishkaMN) [reply lanelet2/lanelet2_core/src/LaneletMap.cpp:804]: Yeah I think we should definitely include that. Because I did forget that adding the primitive as a parameter would register their relation in regulatoryElement layer so it can be queried there.

**Review by msmcconnell — APPROVED** [review]
